### PR TITLE
Module/stringtie/1.0

### PR DESCRIPTION
**Pull request by kmcoyle**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [x ] I used the cookiecutter template and updated the placeholder rules.

- [x ] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [x ] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [ x] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [ x] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [ x] I updated the final target rule (`*_all`) to include every output rule.

- [ x] I explained important module design decisions in `CHANGELOG.md`.

- [ x] I tested the module on real data for all supported `seq_type` values.

- [ x] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [ x] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [ x] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [ x] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [ x] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [ ] I added more granular output subdirectories.

- [ ] I added rules to the `reference_files` workflow to generate any new reference files.

- [ ] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [ ] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

Important! If you are updating the module version, ensure the previous version of the module is restored from master.
If you want to restore a deleted file or directory from the remote master, you can use `git checkout origin/master path/to/file`,
then a `git commit` will ensure that file is tracked on your branch again.
Example:
```
mv modules/strelka/1.1 modules/strelka/1.2
git checkout origin/master modules/strelka/1.1
```

### CODE REVIEW

File: modules/stringtie/1.0/envs/stringtie.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: stringtie
```
Comment (lkhilton) [inline modules/stringtie/1.0/envs/stringtie.yaml]: Can you please put this file in `lcr-modules/envs` and then create a symlink from the `modules/stringtie/1.0/envs` directory pointing at this one?

File: modules/stringtie/1.0/envs/samtools-1.9.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../envs/samtools/samtools-1.9.yaml
```
Comment (lkhilton) [inline modules/stringtie/1.0/envs/samtools-1.9.yaml]: You can delete this symlink from your module since it's not used.

File: modules/stringtie/1.0/stringtie.smk [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    _stringtie_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _stringtie_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
```
Comment (lkhilton) [inline modules/stringtie/1.0/stringtie.smk:56]: I recommend including the `.bai` file as a second input here. The STAR module creates the index in a separate rule, and if one tried to chain these modules together this rule could try to run before the index is created and error out.

File: modules/stringtie/1.0/config/default.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+lcr-modules:
+
+    stringtie_run:
+
+        inputs:
+            # Available wildcards: {seq_type} {genome_build} {sample_id}
+            sample_bam: "__UPDATE__"
```
Comment (lkhilton) [inline modules/stringtie/1.0/config/default.yaml:8]: Does this module _require_ STAR bam files? If yes, perhaps consider renaming to `star_bam` or include a comment here reminding users that only STAR is acceptable.

File: modules/stringtie/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [1.0] - 2021-11-30
+
+This release was authored by Krysta Coyle.
+
+- awk script necessary to introduce XS tags into STAR-aligned BAMs. Without XS tags, stringtie generates single-exon transcripts.
+- GTF mandatory, highly recommended for well-annotated genomes.
```
Comment (lkhilton) [inline modules/stringtie/CHANGELOG.md]: I'm not sure what "GTF mandatory, highly recommended" means... Is this related to the output format? Is the output format modifiable?

**Review by lkhilton — CHANGES_REQUESTED** [review]
I have a few nit-picky requests. Overall looks excellent and we can probably get it approved and merged today!

File: modules/stringtie/1.0/stringtie.smk [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    group:
```
Comment (Kdreval) [inline modules/stringtie/1.0/stringtie.smk:61]: Thank you Krysta, very good module!!! I would chime into the review suggesting to drop the grouping here since it is not used in the module.

File: modules/stringtie/1.0/src/tagXSstrandedData.awk [hunk 7]
```diff
@@ -0,0 +1,41 @@
+## Authored by Alex Dobin, available at https://github.com/alexdobin/STAR/tree/master/extras/scripts
```
Comment (Kdreval) [inline modules/stringtie/1.0/src/tagXSstrandedData.awk:1]: If this script is not modified, should we consider downloading it within the module, since this is not explicitly authored by us? If you are concerned about possible future script updates that would break the functionality, it is possible tor refer to a specific commit (instead of master) to ensure exact same version is always used

File: modules/stringtie/1.0/config/default.yaml (hunk 4 above)
Comment (kmcoyle) [reply modules/stringtie/1.0/config/default.yaml:8]: Updated with comment re: STAR bams.

File: modules/stringtie/1.0/stringtie.smk (hunk 3 above)
Comment (kmcoyle) [reply modules/stringtie/1.0/stringtie.smk:56]: Thanks for pointing that out! Done & added to the demo files.

File: modules/stringtie/1.0/envs/stringtie.yaml (hunk 1 above)
Comment (kmcoyle) [reply modules/stringtie/1.0/envs/stringtie.yaml]: Done!

File: modules/stringtie/1.0/envs/samtools-1.9.yaml (hunk 2 above)
Comment (kmcoyle) [reply modules/stringtie/1.0/envs/samtools-1.9.yaml]: Thanks for catching it!

File: modules/stringtie/CHANGELOG.md (hunk 5 above)
Comment (kmcoyle) [reply modules/stringtie/CHANGELOG.md]: I see how this was confusing. Updated to specify 'reference GTF'.

File: modules/stringtie/1.0/src/tagXSstrandedData.awk (hunk 7 above)
Comment (lkhilton) [reply modules/stringtie/1.0/src/tagXSstrandedData.awk:1]: I thought about this as well but doing it this way is probably the most robust and future-proof and it's still attributed to the repository/author.

File: modules/stringtie/1.0/stringtie.smk (hunk 6 above)
Comment (lkhilton) [reply modules/stringtie/1.0/stringtie.smk:61]: I actually would recommend keeping the grouping but adding it to the _stringtie_run rule - it's meant to group the fast-running input job with the longer-running job to prevent overwhelming the head node or gphost with more jobs than allowed per user.

Comment (Kdreval) [reply modules/stringtie/1.0/stringtie.smk:61]: yes, sure, thanks Laura!

File: modules/stringtie/1.0/src/tagXSstrandedData.awk (hunk 7 above)
Comment (Kdreval) [reply modules/stringtie/1.0/src/tagXSstrandedData.awk:1]: this makes sense, let's keep it as-is 😄

File: modules/stringtie/1.0/stringtie.smk (hunk 6 above)
Comment (kmcoyle) [reply modules/stringtie/1.0/stringtie.smk:61]: Moved the grouping to the _stringtie_run rule

**Review by lkhilton — APPROVED** [review]
